akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -244,15 +272,18 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
           become(Idle)
         }
         def connectionWindowAvailable(): Unit = () // nothing to do, as there is no data to send
-        def enqueueOutStream(outStream: OutStream): Unit =
+        def enqueueOutStream(outStream: OutStream[_]): Unit =
           if (connectionWindowLeft == 0) become(WaitingForConnectionWindow(immutable.TreeSet(outStream.streamId)))
           else {
             require(outStream.canSend)
 
             val maxBytesToSend = currentMaxFrameSize min connectionWindowLeft
             val frame = outStream.nextFrame(maxBytesToSend)
             outlet.push(frame)
-            connectionWindowLeft -= frame.payload.size
+            frame match {
```

Comment:
I guess one consequence of putting it like this is that we will wait for space in the window before sending the trailing `HEADER` frame while that wouldn't be necessary (because control frames are not accounted for with windows).